### DISCUSSION TITLE
[@mantine/core] Allow global change of font-weight for components

### DISCUSSION THREAD

**Discussion by wuifdesign** [mantinedev/mantine#5006] [Feature requests]
### What package has an issue

@mantine/core

### Describe the bug

It is not possible to globally set the font-weight for normal/bold fonts.

i.e. for Button they are hardcoded like this:
```
font-weight: 600;
```

it is currently possible to override each component using the provider with following syntax, but i think this can be done nicer
```
    Button: Button.extend({
      defaultProps: {
        color: 'action',
        fw: 800,
      },
    }),
``` 

### What version of @mantine/* packages do you have in package.json? (Note that all @mantine/* packages must have the same version in order to work correctly)

7.1.2

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

Would be nice to have a config setting for a css variable to override normal/bold font-weights for all components.

like: 
```
--mantine-font-weight-thin: 200;
--mantine-font-weight-normal: 400;
--mantine-font-weight-bold: 600;
```

**Comment by rtivital** [comment]
You can override styles per component with Styles API on theme. Additionally you can apply styles to all Mantine components with `[class^="mantine"]` selector.

**Reply by pkuczynski** [reply]
Unfortunately there is no way to override base font-weight used by all components. It would be nice to be able to do it like we have for headings.
